lib/util/fingerprint.spec.ts

```diff
@@ -0,0 +1,26 @@
+import { fingerprint } from './fingerprint';
+
+describe('util/fingerprint', () => {
+  const obj: any = {
+    name: 'object',
+    type: 'object',
+    isObject: true,
+  };
+
+  const obj2: any = {
+    type: 'object',
+    name: 'object',
+    isObject: true,
+  };
+
+  it('returns empty string', () => {
+    const res = fingerprint(undefined);
+    expect(res).toBeEmptyString();
+  });
+
+  it('maintains deterministic order', () => {
+    const res = fingerprint(obj);
+    const res2 = fingerprint(obj2);
+    expect(Object.keys(res)).toEqual(Object.keys(res2));
```

Comment:
```suggestion
    expect(res).toEqual(res2);
```
i think you like to compare the strings, not the keys of the string object 🤔